plugin/vmss_processor.go

```diff
@@ -0,0 +1,237 @@
+			return nil
+		})
+	}
+
+	// Send instance jobs
+	go func() {
+		defer close(instanceJobs)
+		for _, vmssvm := range vmssvms.Values() {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+
+			instanceJobs <- vmssInstanceJob{
+				vmssvm:        vmssvm,
+				resourceGroup: resourceGroup,
+				vmssName:      *vmss.Name,
+			}
+		}
+	}()
+
+	return g.Wait()
+}
+
+func processVMSSInstanceParallel(ctx context.Context, vmssvm compute.VirtualMachineScaleSetVM,
+	resourceGroup, vmssName string, clients *azureClients,
+	vmToNetworkMap map[string]networkInfo, mu *sync.Mutex) error {
+
+	if vmssvm.InstanceID == nil {
+		return fmt.Errorf("instance ID is nil")
+	}
+
+	// Check if instance is running
+	if running, err := isVMSSInstanceRunning(ctx, clients.vmssvmClient, resourceGroup,
+		vmssName, vmssvm.InstanceID); err != nil {
+		return err
+	} else if !running {
+		return nil
+	}
+
+	// Process network interfaces for the instance
+	var netInfo networkInfo
+	if err := processVMSSNetworkInterfaces(ctx, vmssvm, resourceGroup, vmssName, clients, &netInfo); err != nil {
+		return err
+	}
+
+	func() {
+		mu.Lock()
+		defer mu.Unlock()
+		vmToNetworkMap[*vmssvm.ID] = netInfo
+	}()
+
+	return nil
+}
+
+// VMSS Instance Status Check
+func isVMSSInstanceRunning(ctx context.Context, vmssvmClient *compute.VirtualMachineScaleSetVMsClient,
+	resourceGroup, vmssName string, instanceID *string) (bool, error) {
+
+	if instanceID == nil {
+		return false, fmt.Errorf("instance ID is nil")
+	}
+
+	iv, err := vmssvmClient.GetInstanceView(ctx, resourceGroup, vmssName, *instanceID)
+	if err != nil {
+		return false, fmt.Errorf("error fetching instance view for instance %s: %w", *instanceID, err)
+	}
+	if iv.Statuses == nil {
+		return false, fmt.Errorf("instance view statuses returned for instance %s is null", *instanceID)
+	}
+
+	for _, s := range *iv.Statuses {
+		if s.Code == nil {
+			continue
+		}
+		state := strings.ToLower(*s.Code)
+		prefix := "powerstate/"
+		if !strings.HasPrefix(state, prefix) {
+			continue
+		}
+		state = strings.TrimPrefix(state, prefix)
+		if state == "running" {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// Helper Functions
+func getSetForVMSSInstance(in string) string {
+	trimmed := strings.Split(in, "/"+constVirtualMachinesResource)[0]
+	return trimmed
+}
+
+func getExternalNameforVMSSInstance(in string) (string, error) {
+	splitId := strings.Split(strings.TrimLeft(in, "/"), "/")
+	if len(splitId) != 10 ||
+		!strings.EqualFold(splitId[0], constSubscriptions) ||
+		!strings.EqualFold(splitId[2], constResourceGroups) ||
+		!strings.EqualFold(splitId[4], constProviders) ||
+		!strings.EqualFold(splitId[5], constMsComputeService) ||
+		!strings.EqualFold(splitId[6], constVirtualMachineScaleSetsResource) ||
+		!strings.EqualFold(splitId[8], constVirtualMachinesResource) {
+		return "", fmt.Errorf("unexpected format of virtual machine stateful set ID: %v", splitId)
+	}
+	return splitId[7] + "_" + splitId[9], nil
+}
```

Comment:
I'd still like to see some basic tests for these small string manipulation functions, if you don't mind

Reply:
Yes, of course. I'll be honest, I just forgot to implement them after your last comment on the matter.